lib/net/http/header.rb

```diff
@@ -101,41 +258,63 @@ def add_field(key, val)
     end
   end
 
-  # [Ruby 1.8.3]
-  # Returns an array of header field strings corresponding to the
-  # case-insensitive +key+.  This method allows you to get duplicated
-  # header fields without any processing.  See also #[].
+  # Returns the array field value for the given +key+,
+  # or +nil+ if there is no such field;
+  # see {Fields}[rdoc-ref:Net::HTTPHeader@Fields]:
   #
-  #   p response.get_fields('Set-Cookie')
-  #     #=> ["session=al98axx; expires=Fri, 31-Dec-1999 23:58:23",
-  #          "query=rubyscript; expires=Fri, 31-Dec-1999 23:58:23"]
-  #   p response['Set-Cookie']
-  #     #=> "session=al98axx; expires=Fri, 31-Dec-1999 23:58:23, query=rubyscript; expires=Fri, 31-Dec-1999 23:58:23"
+  #   req = Net::HTTP::Get.new(uri)
+  #   req[:foo] = 'bar'
+  #   req.get_fields(:foo)    # => ["bar"]
+  #   req.add_field(:foo, 'baz')
+  #   req.get_fields(:foo)    # => ["bar", "baz"]
+  #   req.get_fields(:nosuch) # => nil
   #
   def get_fields(key)
     stringified_downcased_key = key.downcase.to_s
     return nil unless @header[stringified_downcased_key]
     @header[stringified_downcased_key].dup
   end
 
-  # Returns the header field corresponding to the case-insensitive key.
-  # Returns the default value +args+, or the result of the block, or
-  # raises an IndexError if there's no header field named +key+
-  # See Hash#fetch
+  # :call-seq
+  #   fetch(key, default_val = nil) {|key| ... }
```

Comment:
I think we should have a call-seq without the block to show that the block is optional.

Reply:
There are many such places thoughout the core, where the call-seq shows the block and the text below says stuff like "Returns an enumerator if no block is given."  Should we amend the doc guide, one way or the other?

Reply:
Well for example, here's the call-seq for Hash#each:

```
each {|key, value| ... } → self
each → new_enumerator
```

And here's the one for Hash#fetch (which is what this is calling internally):

```
fetch(key) → object
fetch(key, default_value) → object
fetch(key) {|key| ... } → object
```

So I think we should have a similar call-seq as `Hash#fetch`.

Reply:
Done.